js/binancetr.js

```diff
@@ -0,0 +1,308 @@
+        //                            {"filterType":"MIN_NOTIONAL","avgPriceMins":"5","minNotional":"10.00000000","applyToMarket":true},
+        //                            {"filterType":"ICEBERG_PARTS","applyToMarket":false,"limit":"10"},
+        //                            {"filterType":"MARKET_LOT_SIZE","minQty":"0.00000000","maxQty":"191987.27930555","stepSize":"0.00000000","applyToMarket":false},
+        //                            {"filterType":"MAX_NUM_ORDERS","applyToMarket":false},
+        //                            {"filterType":"MAX_NUM_ALGO_ORDERS","applyToMarket":false,"maxNumAlgoOrders":"5"}
+        //                        ],
+        //                        "orderTypes":[
+        //                            "LIMIT",
+        //                            "LIMIT_MAKER",
+        //                            "MARKET",
+        //                            "STOP_LOSS_LIMIT",
+        //                            "TAKE_PROFIT_LIMIT"
+        //                            ],
+        //                        "icebergEnable":1,
+        //                        "ocoEnable":1,
+        //                        "spotTradingEnable":1,
+        //                        "marginTradingEnable":0,
+        //                        "permissions":["SPOT"]
+        //                    },
+        //                ]
+        //            }
+        //        }
+        //
+        const result = [];
+        const data = this.safeValue (response, 'data', {});
+        const list = this.safeValue (data, 'list', []);
+        for (let i = 0; i < list.length; i++) {
+            const market = list[i];
+            // const type = this.safeInteger(market, 'type'); // Important to see endpoint for this market
+            const id = this.safeString (market, 'symbol');
+            const filters = this.safeValue (market, 'filters');
+            const filter0 = this.safeValue (filters, 0);
+            const filter2 = this.safeValue (filters, 2);
+            const permissions = this.safeValue (market, 'permissions');
+            const minPrice = this.safeValue (filter0, 'minPrice');
+            const maxPrice = this.safeValue (filter0, 'maxPrice');
+            const minQty2 = this.safeValue (filter2, 'minQty');
+            const maxQty2 = this.safeValue (filter2, 'maxQty');
+            const baseId = this.safeString (market, 'baseAsset');
+            const quoteId = this.safeString (market, 'quoteAsset');
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            const spotPer = permissions.indexOf ('SPOT') > -1;
+            const marginPer = permissions.indexOf ('MARGIN') > -1;
+            const marginEnable = (this.safeInteger (market, 'marginTradingEnable')) === '1' ? true : false;
+            const spotEnable = (this.safeInteger (market, 'spotTradingEnable')) === '1' ? true : false;
```

Comment:
Lines 182-183 violate this rule: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#using-ternary-conditionals

Reply:
Fixed.